tsdb/chunkenc/chunk.go

```diff
@@ -69,6 +73,7 @@ type Chunk interface {
 // Appender adds sample pairs to a chunk.
 type Appender interface {
 	Append(int64, float64)
+	AppendHistogram(h histogram.SparseHistogram)
```

Comment:
I got second thoughts about the signature here (and subsequently the SparseHistogram struct). It would look more consistent if this was `AppendHistogram(ts int64, h histogram.SparseHistogram)`. But it's not just about the look. Everything in `SparseHistogram` _except_ the timestamp comes from the exposition. I'm currently working on the scrape code, and it would make much more sense to get a `SparseHistogram` from the exposition format parser, which then is combined with the current timestamp in the `AppendHistogram` call.

So I'd suggest to remove the timestamp from `SparseHistogram`. @codesome what do you think?

Reply:
Sounds good to me. In case we have more data types in future, having the signature to be `AppendNewDataType(ts, NewDataType)` sounds like a good idea without expecting the value to have the timestamp.